### ISSUE TITLE
The projects' of a Budget get scrambled when browsing from first to second page.

### ISSUE THREAD

**Issue by koenigworks9** [consul/consul#1612]
Hi there!

From the user point of view,  when voting a Budget campaing in the list of available projects, **some projects from the first page gets repeated** in the second page.

I believe the reason is that the first (main) page is linked/referenced without a random_seed, while the second one it has.  For example:

Email sent to the user has the following link:
https://decide.madrid.es/presupuestos/presupuestos-participativos-2017/toda-la-ciudad/toda-la-ciudad?filter=selected&pk_campaign=2017_presupuestos_votacion&pk_source=bbdd_ciudadanos&pk_medium=email&pk_content=link_ciudad_footer

Once inside the page, after voting, the user moves on to the second page:
https://decide.madrid.es/presupuestos/presupuestos-participativos-2017/toda-la-ciudad/1?filter=selected&page=2&pk_campaign=2017_presupuestos_votacion&pk_content=link_ciudad_footer&pk_medium=email&pk_source=bbdd_ciudadanos&random_seed=0.43

Notice that the latter includes **page=2** and **random_seed=0.43** while the former link does not. I believe this might be the reason for some projects to be scrambled and repeated in the second page.

**Comment by iagirre** [comment]
I'll look at this.

**Comment by iagirre** [comment]
I think that there is still some kind of problem when using random order and pagination, even if the random_seed is set. The problem is not the seed in params, because although it's not present at the first link, it is set before index call, and used in controller.
I realised that the investments random selection for the each page changes when the loop is made.
e.g. the @investments_id variable contains [61, 70] but the investment listed on screen (the ones in the loop) are [59, 66], and I think that this can cause the duplicate records.
The problem is that I wasn't able to figure out what's going on there, and the duplications doesn't occur always, so is a little bit difficult to reach a solution.
I found that a semi-random selection can be made using [this method from Stackoverflow](https://stackoverflow.com/a/46527029/1799880) (I think that the problem from that answer is what we need). [I started to implement](https://github.com/wairbut-m2c/consul/tree/iagirre-budget-pagination-bug) and it works, but it needs fixes to make the tests pass and I'm not sure if it's a valid solution. If so, I can finish it.
